File: src/test/java/com/jos/dem/vetlog/validator/ChangePasswordValidatorTest.java
package com.jos.dem.vetlog.validator;

import com.jos.dem.vetlog.command.ChangePasswordCommand;
import lombok.extern.slf4j.Slf4j;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestInfo;
import org.springframework.validation.Errors;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;

@Slf4j
class ChangePasswordValidatorTest {

  private ChangePasswordValidator validator = new ChangePasswordValidator();

  @Test
  @DisplayName("not validating since password does not match")
  void shouldRejectSincePasswordsAreNotEquals(TestInfo testInfo) {
    log.info("Running: {}", testInfo.getDisplayName());
    ChangePasswordCommand command = new ChangePasswordCommand();
    command.setToken("token");
    command.setPassword("password");
    command.setPasswordConfirmation("passwords");
    Errors errors = mock(Errors.class);
    validator.validate(command, errors);
    verify(errors).rejectValue("password", "user.error.password.equals");
  }
}

File: src/test/java/com/josdem/vetlog/validator/ChangePasswordValidatorTest.kt
package com.josdem.vetlog.validator

import com.josdem.vetlog.command.ChangePasswordCommand
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.TestInfo
import org.mockito.kotlin.mock
import org.mockito.kotlin.verify
import org.slf4j.LoggerFactory
import org.springframework.validation.Errors

internal class ChangePasswordValidatorTest {
    companion object {
        private val log = LoggerFactory.getLogger(ChangePasswordValidatorTest::class.java)
    }

    private val validator = ChangePasswordValidator()

    @Test
    fun `should reject since passwords are not equals`(testInfo: TestInfo) {
        log.info(testInfo.displayName)
        val command =
            ChangePasswordCommand().apply {
                token = "token"
                password = "password"
                passwordConfirmation = "passwords"
            }
        val errors: Errors = mock()

        validator.validate(command, errors)

        verify(errors).rejectValue("password", "user.error.password.equals")
    }
}
